### PR TITLE
Split prometheus e2e tests

### PR DESCRIPTION
**Pull request by fpetkovski**
## Description

The current Prometheus e2e tests have both sequential and parallel
subtests. The parallel subtests expect a prometheus-operator configured
to run against all namespaces. These tests run before the sequential ones
and block their execution.

This commit splits the two subtests into dedicated workflows so that
they can run in parallel. The outcome is a reduction of the e2e suite
to from 60m to 40m.

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: test/e2e/prometheus_test.go [hunk 1]
```diff
@@ -305,6 +305,7 @@ func createK8sAppMonitoring(
 }
 
 func testPromRemoteWriteWithTLS(t *testing.T) {
+	t.Parallel()
```
Comment (fpetkovski) [inline test/e2e/prometheus_test.go:308]: This test alone takes around 10-12 minutes to run. It has 20 subtests each of which takes around one minute. We should see if we there are redundant subtests that we can remove, or look for a way to optimize them.

**Review by simonpasquier — APPROVED** [review]
Neat! LGTM

File: test/e2e/prometheus_test.go (hunk 1 above)
Comment (simonpasquier) [reply test/e2e/prometheus_test.go:308]: I checked quickly a while ago but my recollection was that all combinations made sense to be tested.

Comment (fpetkovski) [reply test/e2e/prometheus_test.go:308]: sgtm, I'll dive deeper into that test once we merge this PR. Just waiting for @paulfantom to chime as well

**Review by paulfantom — APPROVED** [review]
LGTM, let's just squash commits on merge.

**Pull request merged**